src/anemoi/training/diagnostics/callbacks/__init__.py

```diff
@@ -1081,58 +86,109 @@ def get_callbacks(config: DictConfig) -> list:  # noqa: C901
                             mode="max",
                             **checkpoint_settings,
                         ),
-                    ],
+                    ]
                 )
             else:
                 LOGGER.debug("Not setting up a checkpoint callback with %s", save_key)
     else:
         # the tensorboard logger + pytorch profiler cause pickling errors when writing checkpoints
         LOGGER.warning("Profiling is enabled - will not write any training or inference model checkpoints!")
+    return None
 
-    if any([config.diagnostics.log.wandb.enabled, config.diagnostics.log.mlflow.enabled]):
-        from pytorch_lightning.callbacks import LearningRateMonitor
 
-        trainer_callbacks.append(
-            LearningRateMonitor(
-                logging_interval="step",
-                log_momentum=False,
-            ),
-        )
+def _get_config_enabled_callbacks(config: DictConfig) -> list[Callback]:
+    """Get callbacks that are enabled in the config as according to CONFIG_ENABLED_CALLBACKS
 
-    if config.diagnostics.eval.enabled:
-        trainer_callbacks.append(RolloutEval(config))
+    Provides backwards compatibility
+    """
+    callbacks = []
+
+    def nestedget(conf: DictConfig, key, default):
```

Comment:
This one needs some comments if we keep it. Took me a second that we're recursively finding a key.

Reply:
Fair enough, its a little annoying its not in the `DictConfig` by defaut.